src/r/R/RecordFactory.ts

```diff
@@ -506,7 +506,9 @@ export class RecordFactory<T extends RT> {
       const [type, id] = record.split(":"); // [scene, 1]
       const recordF = new RecordFactory(currentRecord);
       // Number(undefined) = NaN, so this will work. Complexity of checking isNaN and hashmap lookup are same.
-      const child = recordF.getRecord(type as RT, Number(id));
+      const children = recordF.getAllDeepChildrenWithFilter(type as RT, r => r.id === Number(id));
```

Comment:
This is a running in a loop. If there is a scenario like `project -> scene1 -> group1 -> element1`. 
when the loop runs for `group1`, a new instance of RecordFactory is created for that record. element1 is the 1st child in that list. Same goes for any level of nested elements.

Reply:
This shouldn't be required at all.